### PR TITLE
closes #649: remove usage of ABCMeta to prevent performance issues on…

### LINKED ISSUE #649: Performance issue for create object in RhinoPython
**Describe the bug**
Creating object is very slow in RhinoPython. There seems to be a memory leak.
compas.version = '0.17.1' pulled from source e5b1e23484f6951c1c2013dfa58ae16349ddfdd4

**To Reproduce**
This function test3() creates 3 lines, the test runs this 100 times:
```
from compas.geometry import Line

def test3():
    line1 = Line([-100, 0, 0], [100, 0, 0])
    line2 = Line([0, -100, 0], [0, 100, 0])
    line3 = Line([0, -100, 10], [0, 100, 10])

if __name__ == "__main__":
    import datetime

    first_time = datetime.datetime.now()
    for i in range(100):
        test3()
    later_time = datetime.datetime.now()
    difference = later_time - first_time
    print("Time took: %s" % difference)
```
**Running this is cPython 3.7** 
Time took: 0:00:00.004987

**Running this in Rhino (fresh after ResetScriptEngine)**
Time took: 0:00:00.051000
**Subsequent runs:**
Time took: 0:00:00.319000
Time took: 0:00:00.490000
Time took: 0:00:01.155000
Time took: 0:00:01.173000
Time took: 0:00:01.154000
Time took: 0:00:02.093000
Time took: 0:00:02.136000
Time took: 0:00:02.133000
Time took: 0:00:02.648000
you get the idea, it keeps going up... until Rhino never returns (aka crash)

### PR DESCRIPTION
**Pull request by gonzalocasas**
… IronPython

### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas/base.py [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
 
 
 __all__ = [
     'Base',
 ]
+import abc
+ABC = abc.ABCMeta('ABC', (object,), {'__slots__': ()})
 
 
 class Base(ABC):
```
Comment (beverlylytle) [inline src/compas/base.py]: Don't you think the other classes will be jealous that `Base` still inherits from `ABC`?

**Comment by beverlylytle** [comment]
Should there also be a PR on compas_fab to remove ABCMeta from the backend features?

File: src/compas/base.py (hunk 1 above)
Comment (tomvanmele) [reply src/compas/base.py]: they were indeed

**Comment by gonzalocasas** [comment]
Thank you both for the quick review!
@beverlylytle yes, we should do the same for fab.
@brgcode could we please push a new patch release 0.17.2 with this fix?
Thx!

**Comment by tomvanmele** [comment]
done...

**Pull request merged**